lib/biokbase/catalog/registrar.py

```diff
@@ -437,16 +434,18 @@ def validate_method_specs(self, basedir):
                         display_yaml = display_yaml_file.read()
 
                     # gather any extra html files
-                    extraFiles = {}
+                    extra_files = {}
                     for extra_file_name in os.listdir(os.path.join(method_path)):
-                        if not os.path.isfile(os.path.join(method_path,extra_file_name)): break
-                        if not extra_file_name.endswith('.html'): break
-                        with codecs.open(oos.path.join(method_path,extra_file_name), 'r', "utf-8", errors='ignore') as extra_file: 
-                            extrafiles[extra_file_name] = extra_file.read()
+                        if not os.path.isfile(os.path.join(method_path,extra_file_name)):
```

Comment:
The tests just ensure that the extra file doesn't cause the registration to fail. That doesn't really test what's going on here or the 3 paths through the code (or the loop)

Reply:
So the 'if not extra_file_name.endswith('.html'):' gets triggered by the spec.json and display.yaml files but I'll add a directory to the test repo to make sure the last path gets triggered

As I said above, I don't really understand what NMS is doing with the passed files other than storing them and I think testing that functionality should be in NMS tests not here in catalog

Reply:
I could remove this code entirely and the test would still pass. That means the test is not effective.

Reply:
I don't know how to test this any better, maybe we just strip it out. NMS just swallows up the extra_files without a peep. Nothing I can see in the NMS spec even returns "extra_files"

Reply:
It looks to me like the validate_method() call that takes the extra_files parameter doesn't actually make any changes NMS side - it just validates that everything is OK. The extra files end up being used in this 700+ line method: https://github.com/kbase/narrative_method_store/blob/master/src/us/kbase/narrativemethodstore/db/NarrativeMethodData.java#L89-L801

Hence, the only way to really test this is to make it fail based on the content of the extra files, as far as I can tell. 

Given you would have to figure out what that 700+ line method does and how to make it fail with some magic file, I think it's time to give up.

Reply:
So strip out the python functionality?

Reply:
Leave it I suppose.